sort_images.sh

```diff
@@ -0,0 +1,159 @@
+#! /usr/bin/env bash
+
+#----------------------------------------------------------------------
+# NAME: sort_images.sh
+#
+# BRIEF: Command-line Bash script with prompts to help sort photo files.
+# - creates & sorts image files into year & month subdirectories
+# - also creates day subdirectories, if answer at prompt is `Y` for Yes
+# - sorts by photo-taken date or if missing, by modify date - since sometimes
+# creation date is assigned as file download date, which is incorrect.
+#
+# AUTHOR: Kim Lew
+#----------------------------------------------------------------------
+
+if [ $# -eq 0 ]; then
+  echo "Type the directory path that needs Year & Month subdirectories."
+  echo "Leave off trailing / at end of path: "
+  read -r directory_path
+  echo "Do you also want a Day subdirectory? (Y for Yes, N for No):"
+  read -r day_subdir_also
+else
+  if [ ! "$1" ]; then
+    echo "Enter the directory path - as the 1st parameter."
+    echo "Enter Y for Yes or N for No - for Day subdirectory as 2nd parameter."
+    exit 1
+  fi
+fi
+
+if [[ $# -eq 1 || $# -eq 2 ]]; then
```

Comment:
You should try giving your script _three_ arguments, by 'accident', and see what happens.

Reply:
@kcharter Thanks. I forgot to test for that. Addressed it. Let me know if that works.

Reply:
Looks good, and the script now exits with a nice error message when I give more than two arguments. It also still seems to work as expected when I give one or two command-line arguments, or when I give none and just answer the prompts.